.github/scripts/bin_diff.ps1

```diff
@@ -0,0 +1,258 @@
+  [CmdletBinding()]
+  param (
+    [parameter(Mandatory=$TRUE,Position=0,HelpMessage="Path create.")]
+    [string]$Path
+  )
+  PROCESS {
+    New-Item -path $Path -itemtype Directory -force | Out-Null
+  }
+}
+
+# GETS ALL FILES IN A PATH RECURSIVELY #
+function GetFiles {
+  [CmdletBinding()]
+  param (
+    [parameter(Mandatory=$TRUE,Position=0,HelpMessage="Path to get files for.")]
+    [string]$Path
+  )
+  PROCESS {
+    ls $Path -r | where { !$_.PSIsContainer }
+  }
+}
+
+# GETS ALL FILES WITH CALCULATED HASH PROPERTY RELATIVE TO A ROOT DIRECTORY RECURSIVELY #
+# RETURNS LIST OF @{RelativePath, Hash, FullName}
+function GetFilesWithHash {
+  [CmdletBinding()]
+  param (
+    [parameter(Mandatory=$TRUE,Position=0,HelpMessage="Path to get directories for.")]
+    [string]$Path,
+
+    [parameter(HelpMessage="The hash algorithm to use.")]
+    [string]$Algorithm="MD5"
+  )
+  PROCESS {
+    $OriginalPath = $PWD
+    SetWorkDir path/to/diff $Path
+    GetFiles $Path | select @{N="RelativePath";E={$_.FullName | Resolve-Path -Relative}},
+                            @{N="Hash";E={(Get-FileHash $_.FullName -Algorithm $Algorithm | select Hash).Hash}},
+                            FullName
+    SetWorkDir path/to/original $OriginalPath
+  }
+}
+
+filter UpdatedOnlyFilter{
+param(
+        [Parameter(Position=0, Mandatory=$true,ValueFromPipeline = $true)]
+        [ValidateNotNullOrEmpty()]
+        [PSCustomObject]
+        $obj
+    )
+
+    $obj|?{$_.Status -ne '--'}
+}
+
+# COMPARE TWO DIRECTORIES RECURSIVELY #
+# RETURNS LIST OF @{RelativePath, Hash, FullName}
+function DiffDirectories {
+  [CmdletBinding()]
+  param (
+    [parameter(Mandatory=$TRUE,Position=0,HelpMessage="Directory to compare left.")]
+    [alias("l")]
+    [string]$LeftPath,
+
+    [parameter(Mandatory=$TRUE,Position=1,HelpMessage="Directory to compare right.")]
+    [alias("r")]
+    [string]$RightPath
+  )
+  PROCESS {
+    $LeftHash = GetFilesWithHash $LeftPath
+    $RightHash = GetFilesWithHash $RightPath
+    diff -ReferenceObject $LeftHash -DifferenceObject $RightHash -Property RelativePath,Hash | 
+    Select-Object *, @{
+    Name = 'Status'
+    Expression = {
+        if ($_.SideIndicator -eq "=>") {
+          if (-not (Test-Path (Join-Path -Path $LeftPath -ChildPath $_.RelativePath))) {
+                'Added'
+          }
+          else {
+                'Modified'
+          }
+        }
+        if ($_.SideIndicator -eq "<=") {
+          if (-not (Test-Path (Join-Path -Path $RightPath -ChildPath $_.RelativePath))) {
+                'Deleted'
+          }
+          else {
+                '--'
+          }
+        }
+      }
+    } | UpdatedOnlyFilter | Sort-Object Status
+  }
+}
+
+
+### END FUNCTION DEFINITIONS ###
+
+### PROGRAM LOGIC ###
+
+if($Compare.length -ne 2) {
+  Print -x "Compare requires passing exactly 2 path parameters separated by comma, you passed $($Compare.length)." -f
+}
+Print "Comparing $($Compare[0]) to $($Compare[1])..." -a 1
```

Comment:
so I have a bit of a hard time understanding how this code works - but from your screenshot I have a few questions/comments.

1. I think noting that a file is `modified` if it still exists is probably just noise - IMO we don't really care, I expect files to be different when I do a build - they will contain different version numbers etc.
2. What I really care about are new files and deleted files - if we need to keep `modified` files - can we make deleted and added files pop out some how? (color text?)
3. Is a deleted file also considered modified or does it have a separate state? Can you show a screenshot of this?

Reply:
1. Agreed, do you suggest removing modified entries? I think we can let them be.
2. I have added colors to make them pop, green for Added, and red for Deleted entries.
3. Yes, deleted files has a separate "Deleted" label. attaching ss below.
![Screenshot 2023-05-22 at 7 22 26 PM](https://github.com/DynamoDS/Dynamo/assets/32665108/be77060a-815a-44d6-ac77-a1ed007e19e6)

Reply:
removed hash and side indicator columns as they were not useful.

![Screenshot 2023-05-22 at 8 32 56 PM](https://github.com/DynamoDS/Dynamo/assets/32665108/24977811-40d9-4b47-b4e7-6583aeac51c4)